azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/messager/AzureMessager.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.messager;
+
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.operation.IAzureOperation;
+import lombok.RequiredArgsConstructor;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.Optional;
+
+public abstract class AzureMessager implements IAzureMessager {
+    private static IAzureMessager defaultMessager;
+
+    public static synchronized void setDefaultMessager(@Nonnull IAzureMessager messager) {
+        if (AzureMessager.defaultMessager == null) {
+            AzureMessager.defaultMessager = messager;
+        } else {
+            // not allow overwriting...
+            throw new AzureToolkitRuntimeException("default messager has already been registered");
+        }
+    }
+
+    @Nonnull
+    public static IAzureMessager getDefaultMessager() {
```

Comment:
I think it is not supposed to be called from outside, make it private?

Reply:
I am not sure whether we need to use the default messager directly.